### PR TITLE
cleaned up files to accommodate renamings

### PR DESCRIPTION
**Pull request by tenbradley**
Through the last several updates to states, many names changed and things changed location. This pr should go through and fix various imports and such.

Also updates code base to remove circular dependency. Yay!

### CODE REVIEW

File: state/HypermediaState.js [hunk 1]
```diff
@@ -30,14 +33,22 @@ export class HypermediaState extends Fetchable(Object) {
...
-			const basicInfo = sirenComponentBasicInfo(propertyInfo, this);
+			const basicInfo = sirenDefinedProperty(propertyInfo, this);
 			if (!basicInfo) return;
 
 			const sirenComponent = this._getSirenComponent(basicInfo);
 			sirenComponent.addComponent(component, name, { route: basicInfo.route ? { [name]: basicInfo.route } : undefined, method: observables[name].method });
 		});
 	}
 
+	createChildState(entityID, token) {
```
Comment (m6jones) [inline state/HypermediaState.js:44]: Don't need to pass a token :)

File: test/observable/observable.test.js [hunk 2]
```diff
@@ -177,9 +182,9 @@ describe('create observable multiple components', () => {
 
 		const methods = obj._observers._methods;
 		assert.isFalse(methods.has(comp1), 'method for first observer was not removed');
-		assert.equal(comp1['foo'], 'bar', 'first observer maps to wrong value');
-		assert(methods.has(comp2));
-		assert(methods.get(comp2) === method2);
-		assert(comp2['abc'] === 'xyz');
+		assert.equal(comp1['foo'], 'bar1', 'first observer maps to wrong value');
+		assert.isTrue(methods.has(comp2), 'method for second observer should exist');
+		assert.equal(methods.get(comp2), method2, 'method for second observer is incorrect method');
```
Comment (AKobets) [inline test/observable/observable.test.js:187]: be more positive in assert comments, and write what assertion is expecting.
For example: `observer methods map should have method for component`

File: test/observable/observable.test.js [hunk 3]
```diff
@@ -150,7 +153,7 @@ describe('create observable multiple components', () => {
 		const methods = obj._observers._methods;
 		assert.isTrue(methods.has(comp1), 'method for first observer not stored');
 		assert.equal(methods.get(comp1), method1, 'incorrect method stored for first observer');
-		assert.equal(comp1['foo'], 'bar', 'first observer maps tp wrong value');
+		assert.equal(comp1['foo'], 'bar1', 'first observer maps tp wrong value');
```
Comment (AKobets) [inline test/observable/observable.test.js:156]: tpo in `first observer maps tp wrong value`

File: test/observable/observable.test.js [hunk 4]
```diff
@@ -1,185 +1,209 @@
 import { assert }  from '@open-wc/testing';
-import { Component } from '../../state/observable/Common.js';
 import { Observable } from '../../state/observable/Observable.js';
+import { ObserverMap } from '../../state/observable/ObserverMap.js';
 
-describe('Observable methods', () => {
-	it('Observable constructed with Component field', () => {
-		const obj = new Observable();
+describe.only('Observable methods', () => {
```
Comment (zinabat) [inline test/observable/observable.test.js:5]: You left a `.only` in here

File: test/observable/observable.test.js [hunk 5]
```diff
@@ -1,185 +1,209 @@
...
 		assert.equal(map.size, 0, 'Observable has observer after it has been removed');
 
 		const methods = obj._observers._methods;
 		assert.isFalse(methods.has(comp), 'observer method tracked after observer is removed');
-		assert.equal(comp['foo'], 'bar', 'observer maps method correctly after removed from observable');
 	});
 });
 
-describe('create observable multiple components', () => {
+describe.only('create observable multiple components', () => {
```
Comment (zinabat) [inline test/observable/observable.test.js:91]: Same thing here with `.only`

File: test/observable/observable.test.js [hunk 6]
```diff
@@ -1,185 +1,209 @@
...
+		const comp1 = new ObserverMap();
+		const obj = new Observable({});
 
-		obj.addObserver(comp1, 'foo', {}, false);
-		obj.addObserver(comp1, 'abc', {}, 'xyz');
+		obj.addObserver(comp1, 'foo', {});
+		obj.addObserver(comp1, 'abc', {});
 
-		const map = obj._observers.components;
+		const map = obj._observers._observers;
```
Comment (zinabat) [inline test/observable/observable.test.js:149]: I don't love the naming on these things but I understand for now.

File: state/HypermediaState.js (hunk 1 above)
Comment (tenbradley) [reply state/HypermediaState.js:44]: But....why?

**Review by m6jones — APPROVED** [review]

**Pull request merged**